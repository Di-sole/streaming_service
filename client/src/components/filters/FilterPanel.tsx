import Image from 'next/image';
import styles from './FilterPanel.module.scss';
import closeIcon from '../../../public/icons/close.svg';
import { ReactNode, useEffect, useState } from 'react';
import TransparentButton from '../UI/buttons/TransparentButton/TransparentButton';

import { useTranslation } from 'next-i18next';
import { useAppDispatch, useAppSelector } from '../../store/hooks/redux';
import { filtersRemoved } from '../../store/slices/moviesSlice';
interface IPanel {
    className?: string;
    isFilterApplied: boolean;
    children: ReactNode;
}

const FilterPanel = ({ className, children, isFilterApplied }: IPanel) => {
    const { t } = useTranslation('moviesPage');
    const dispatch = useAppDispatch();

    const HandleRemoveFilters = () => {
        dispatch(filtersRemoved({ genres: [], countries: [], ratingKp: 0, votesKp: 0, director: '', actor: '' }));
    };

    return (
        <section className={[styles.container, className].join(' ')}>
            <div className={styles.container__body}>
                <div className={styles.container__content}>
                    <div className={styles.container__filterList}>{children}</div>
                    <div className={styles.container__buttonContainer}>
                        <TransparentButton
                            className={[
                                styles.container__resetButton,
                                !isFilterApplied ? styles.container__resetButton_disabled : null,
                            ].join(' ')}
                            textColor="bright"
                            onClick={HandleRemoveFilters}
                        >
                            <Image src={closeIcon} height={16} width={16} alt="Иконка-крестик" />
                            {t('filterPanel.removeFilters')}
                        </TransparentButton>
                    </div>
                </div>
            </div>
        </section>
    );
};

export default FilterPanel;
