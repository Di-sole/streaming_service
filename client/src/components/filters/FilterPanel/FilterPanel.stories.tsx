import type { Meta, StoryObj } from '@storybook/react';
import Panel from './FilterPanel';
import './FilterPanel.module.scss';
import FilterPlank from '../FilterPlank/FilterPlank';
import plankStyles from '../FilterPlank/FilterPlank.module.scss';
import FilterList from '../FilterList/FilterList';
import '../FilterList/FilterList.module.scss';
import FilterRange from '../FilterRange/FilterRange';
import '../FilterRange/FilterRange.module.scss';
import FilterSearch from '../FilterSearch/FilterSearch';
import '../FilterSearch/FilterSearch.module.scss';
import { genres, countries, actorSuggestions, directorSuggestions } from '../temp/items';
import rating from '../../../../public/icons/rating.svg';
import rank from '../../../../public/icons/userRank.svg';

const meta: Meta<typeof Panel> = {
    title: 'Filters/FilterPanel',
    component: Panel,
};

export default meta;

type Story = StoryObj<typeof Panel>;

export const FilterPanel: Story = {
    args: {
        isFilterApplied: true,
        children: (
            <>
                <FilterPlank title="Жанры" className={plankStyles.container__dropdown_leftPositioned}>
                    <FilterList category="genres" items={genres} />
                </FilterPlank>
                <FilterPlank title="Страны" className={plankStyles.container__dropdown_centerPositioned}>
                    <FilterList category="countries" items={countries} />
                </FilterPlank>
                <FilterPlank title="Рейтинг" className={plankStyles.container__dropdown_narrow}>
                    <FilterRange category="ratingKp" image={rating} limit={10} step={0.1} />
                </FilterPlank>
                <FilterPlank title="Оценки" className={plankStyles.container__dropdown_narrow}>
                    <FilterRange category="votesKp" image={rank} limit={1000000} step={1000} />
                </FilterPlank>
                <FilterPlank title="Режиссер" className={plankStyles.container__dropdown_narrow}>
                    <FilterSearch category="director" suggestionsList={directorSuggestions} />
                </FilterPlank>
                <FilterPlank title="Актер" className={plankStyles.container__dropdown_narrow}>
                    <FilterSearch category="actor" suggestionsList={actorSuggestions} />
                </FilterPlank>
            </>
        ),
    },
};
