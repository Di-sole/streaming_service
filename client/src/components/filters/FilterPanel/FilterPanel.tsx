import Image from 'next/image';
import styles from './FilterPanel.module.scss';
import closeIcon from '../../../../public/icons/close.svg';
import { ReactNode } from 'react';
import TransparentButton from '../../UI/buttons/TransparentButton/TransparentButton';

import { useTranslation } from 'next-i18next';
import { useAppDispatch } from '../../../store/hooks/redux';
import { filtersRemoved } from '../../../store/slices/moviesSlice';

interface IPanel {
    className?: string;
    isFilterApplied: boolean;
    children: ReactNode;
}

const FilterPanel = ({ className, children, isFilterApplied }: IPanel) => {
    const { t } = useTranslation('moviesPage');
    const dispatch = useAppDispatch();

    const handleRemoveFilters = () => {
        dispatch(filtersRemoved({ genres: [], countries: [], ratingKp: 0, votesKp: 0, director: '', actor: '' }));
    };

    return (
        <div className={[styles.container, className].join(' ')} data-testid={'filterPanel'}>
            <div className={styles.container__body}>
                <div className={styles.container__content}>
                    <div className={styles.container__filterList}>{children}</div>
                    <div className={styles.container__buttonContainer}>
                        <TransparentButton
                            className={[
                                styles.container__resetButton,
                                !isFilterApplied ? styles.container__resetButton_disabled : null,
                            ].join(' ')}
                            textColor="bright"
                            onClick={handleRemoveFilters}
                        >
                            <Image src={closeIcon} height={16} width={16} alt="Иконка-крестик" />
                            {t('filterPanel.removeFilters')}
                        </TransparentButton>
                    </div>
                </div>
            </div>
        </div>
    );
};

export default FilterPanel;
