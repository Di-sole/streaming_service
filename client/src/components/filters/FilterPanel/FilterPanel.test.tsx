import { render, screen } from '@testing-library/react';
import FilterPanel from './FilterPanel';

jest.mock('../../../store/hooks/redux');
jest.mock('next-i18next', () => ({
    useTranslation: () => {
        return {
            t: (str: string) => str,
            i18n: {
                changeLanguage: () => new Promise(() => {}),
            },
        };
    },
}));

describe('FILTERPANEL SNAPSHOTS TESTS', () => {
    test('FilterPanel should not be changed', () => {
        render(<FilterPanel children="Фильтры" isFilterApplied={false} />);

        const section = screen.getByTestId('filterPanel');
        expect(section).toMatchSnapshot();
    });
});
