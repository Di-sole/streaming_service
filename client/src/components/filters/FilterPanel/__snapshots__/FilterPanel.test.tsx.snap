// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FILTERPANEL SNAPSHOTS TESTS FilterPanel should not be changed 1`] = `
<section
  class="container "
  data-testid="filterPanel"
>
  <div
    class="container__body"
  >
    <div
      class="container__content"
    >
      <div
        class="container__filterList"
      >
        Фильтры
      </div>
      <div
        class="container__buttonContainer"
      >
        <button
          class="button button_bright container__resetButton container__resetButton_disabled"
          type="button"
        >
          <img
            alt="Иконка-крестик"
            data-nimg="1"
            decoding="async"
            height="16"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=32&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=16&q=75 1x, /_next/image?url=%2Fimg.jpg&w=32&q=75 2x"
            style="color: transparent;"
            width="16"
          />
          filterPanel.removeFilters
        </button>
      </div>
    </div>
  </div>
</section>
`;
